core/services/gasupdater/gas_updater.go

```diff
@@ -365,3 +364,21 @@ func (gu *gasUpdater) setPercentileGasPrice(gasPrice int64) error {
 func (gu *gasUpdater) RollingBlockHistory() []Block {
 	return gu.rollingBlockHistory
 }
+
+func isUsableTx(tx Transaction, minGasPriceWei *big.Int) bool {
+	// GasLimit 0 is impossible on Ethereum official, but IS possible
+	// on forks/clones such as RSK. We should ignore these transactions
+	// if they come up since they are not normal.
+	if tx.GasLimit == 0 {
+		return false
+	}
+	// GasPrice 0 on most chains is great since it indicates cheap/free transctions.
+	// However, xDai reserves a special type of "bridge" transaction with 0 gas
+	// price that is always processed at top priority. Ordinary transactions
+	// must be priced at least 1GWei, so we have to discard anything priced
+	// below that (unless the contract is whitelisted).
+	if tx.GasPrice.Cmp(minGasPriceWei) < 0 {
```

Comment:
This actually could throw off the statistics since it means we would take a smaller sample, ignoring potentially useful cheap transactions.

The only alternative I see is to add xDai specific logic here, anyone see another approach?

Reply:
I think we'll probably need chain specific logic in the gasupdater regardless as we add more chains (solana, avalanche etc.) so seems fine to rip off that bandaid now

Reply:
You could probably get fairly far though with keeping a map[chainID]includeTx where includeTx is just a function with the chain specific logic on whether you include the tx in the stats or not